pkg/cloud/services/compute/instance.go

```diff
@@ -639,6 +650,27 @@ func (s *Service) DeleteInstance(eventObject runtime.Object, openStackMachineSpe
 	return s.deleteInstance(eventObject, instanceStatus.InstanceIdentifier())
 }
 
+func (s *Service) ensureTrunkDeletion(eventObject runtime.Object, portID string) error {
```

Comment:
The `s.networkingService.DeleteTrunk` already uses `util.PollImmediate` to retry deletion.
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/8ae251502dbbcdd5685521272c7b4a5659152786/pkg/cloud/services/networking/trunk.go#L92-L104

What is the difference between the retry here and in the snippet above? Maybe it would be sufficient to add `capoerrors.IsConflict` to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/8ae251502dbbcdd5685521272c7b4a5659152786/pkg/cloud/services/networking/trunk.go#L98 if needed.

Reply:
Thanks, will check that.

Reply:
Changes made and waiting for the e2e test result